### ISSUE TITLE
Failed to send request to webcast/sign_url/

### ISSUE THREAD

**Issue by achyutkneupane** [Davincible/gotiktoklive#7]
I am getting this error message when fetching live-room information.
```bash
Started polling
```
```bash
panic: Failed to send request to im/fetch/: Failed to sign request: Failed to send request to webcast/sign_url/: you have exceeded the rate limit, please wait a few min
```

**Comment by Davincible** [comment]
@achyutkneupane  Yes, we are aware of it. TikTok has pushed out an update that broke the API. We are working on a fix but nothing has been found yet.

**Comment by Thecreator1** [comment]
<h1 style="box-sizing: inherit; -webkit-font-smoothing: antialiased; word-break: break-word; overflow-wrap: break-word; border: none; font-size: 3.6rem; line-height: 1.2; margin: 32px 0px; padding: 0px; text-align: left; font-family: &quot;Sofia Pro&quot;, PingFangSC, sans-serif; font-weight: 700; color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: -0.27px; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Rate Limits on API Requests</h1><p style="box-sizing: inherit; -webkit-font-smoothing: antialiased; word-break: break-word; overflow-wrap: break-word; border: none; font-size: 18px; line-height: 1.5; margin: 0px 0px 14px; padding: 0px; text-align: left; color: rgb(0, 0, 0); font-family: &quot;Sofia Pro&quot;, PingFangSC, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: -0.27px; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">TikTok API limits the number of requests you can send in a given timeframe. Limits for each API are set and enforced separately. Default limits are provided below.</p>

API | Limit
-- | --
/v2/user/info/ | 600
/v2/video/query/ | 600
/v2/video/list/ | 600

<p style="box-sizing: inherit; -webkit-font-smoothing: antialiased; word-break: break-word; overflow-wrap: break-word; border: none; font-size: 18px; line-height: 1.5; margin: 0px 0px 14px; padding: 0px; text-align: left; color: rgb(0, 0, 0); font-family: &quot;Sofia Pro&quot;, PingFangSC, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: -0.27px; orphans: 2; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Request rate calculation is based on a one minute sliding window. If the number of requests exceeds the threshold, new requests will be throttled and a response will be returned with HTTP status<span> </span><code style="box-sizing: inherit; -webkit-font-smoothing: antialiased; word-break: break-word; overflow-wrap: break-word; border: none; font-size: 16.2px; line-height: inherit; margin: 0px 0px 15px; padding: 0px 1px; text-align: inherit; font-family: monospace; background: rgb(218, 221, 225); border-radius: 2px;">429</code><span> </span>and error code<span> </span><code style="box-sizing: inherit; -webkit-font-smoothing: antialiased; word-break: break-word; overflow-wrap: break-word; border: none; font-size: 16.2px; line-height: inherit; margin: 0px 0px 15px; padding: 0px 1px; text-align: inherit; font-family: monospace; background: rgb(218, 221, 225); border-radius: 2px;">rate_limit_exceeded</code>.</p>
